src/manifest_store.rs

```diff
@@ -200,43 +213,74 @@ impl ManifestStore {
         Ok(())
     }
 
-    pub(crate) async fn read_latest_manifest(
+    /// Delete a manifest from the object store.
+    pub(crate) async fn delete_manifest(&self, id: u64) -> Result<(), SlateDBError> {
+        // TODO Once we implement snapshots, we should check if the manifest is snapshotted as well
+        let (active_id, _) = self
+            .read_latest_manifest()
+            .await?
+            .ok_or(SlateDBError::ManifestMissing)?;
+        if active_id == id {
+            return Err(SlateDBError::InvalidDeletion);
+        }
+        let manifest_path = &self.get_manifest_path(id);
+        self.object_store.delete(manifest_path).await?;
+        Ok(())
+    }
+
+    /// Read a manifest from the object store.
```

Comment:
nit: specify that the last element of the list is considered the "current" manifest.

Reply:
315be41c00128770b9e3826a811bb34bfbcbed66